### PR TITLE
Setup contract structure, add NFT Opt-in method

### PR DESCRIPTION
**Pull request by D0rs4n**
- Add NFT opt-in method, which takes a deposit PaymentTXN and then records that the contract opted into a creators NFT, so that it can be further processed and fractionalized.

### CODE REVIEW

File: smart_contracts/fractdistribution/contract.py [hunk 1]
```diff
@@ -22,5 +28,86 @@ def delete() -> pt.Expr:
...
+        (creator := pt.AssetParam.creator(pt.Txn.assets[0])),
+        pt.Assert(creator.hasValue()),
+        pt.Assert(creator.value() == pt.Txn.sender()),
+        pt.Assert(deposit_payment_txn.get().sender() == pt.Txn.sender()),
+        pt.Assert(
+            deposit_payment_txn.get().receiver()
+            == pt.Global.current_application_address()
+        ),
+        # the Deposit is 5 Algos = 5000000 microAlgos
+        pt.Assert(deposit_payment_txn.get().amount() == pt.Int(5000000)),
```
Comment (visdauas) [inline smart_contracts/fractdistribution/contract.py:101]: Minimum balance for opt-in needs 100,000 microalgo

File: smart_contracts/fractdistribution/contract.py [hunk 2]
```diff
@@ -1,8 +1,14 @@
 import beaker
 import pyteal as pt
 from algokit_utils import DELETABLE_TEMPLATE_NAME, UPDATABLE_TEMPLATE_NAME
+from .data_utils import MappingState, FracticNFTPool
 
-app = beaker.Application("fractdistribution")
+
+app = beaker.Application(
+    "fractdistribution",
```
Comment (visdauas) [inline smart_contracts/fractdistribution/contract.py:8]: typo

File: smart_contracts/fractdistribution/contract.py [hunk 3]
```diff
@@ -22,5 +28,86 @@ def delete() -> pt.Expr:
...
+                pt.TxnField.type_enum: pt.TxnType.AssetConfig,
+                pt.TxnField.config_asset_default_frozen: pt.Int(0),
+                pt.TxnField.config_asset_unit_name: pt.Bytes("FRC"),
+                pt.TxnField.config_asset_manager: pt.Global.current_application_address(),
+                pt.TxnField.config_asset_reserve: pt.Global.current_application_address(),
+                pt.TxnField.config_asset_clawback: pt.Global.current_application_address(),
+                pt.TxnField.config_asset_total: max_fraction.get(),
+                pt.TxnField.config_asset_decimals: pt.Int(0),
+            }
+        ),
```
Comment (visdauas) [inline smart_contracts/fractdistribution/contract.py:64]: Include the url and metadata of the original asset here

**Review by visdauas — CHANGES_REQUESTED** [review]

File: smart_contracts/fractdistribution/contract.py (hunk 1 above)
Comment (D0rs4n) [reply smart_contracts/fractdistribution/contract.py]: Done

File: smart_contracts/fractdistribution/contract.py (hunk 2 above)
Comment (D0rs4n) [reply smart_contracts/fractdistribution/contract.py]: Done

File: smart_contracts/fractdistribution/contract.py (hunk 3 above)
Comment (D0rs4n) [reply smart_contracts/fractdistribution/contract.py]: Done..

File: smart_contracts/fractdistribution/contract.py [hunk 4]
```diff
@@ -60,6 +64,8 @@ def init_fractic_nft_flow(
                 pt.TxnField.config_asset_clawback: pt.Global.current_application_address(),
                 pt.TxnField.config_asset_total: max_fraction.get(),
                 pt.TxnField.config_asset_decimals: pt.Int(0),
+                pt.TxnField.config_asset_url: original_asset_url.value(),
+                pt.TxnField.config_asset_metadata_hash: original_asset_hash.value(),
```
Comment (visdauas) [inline smart_contracts/fractdistribution/contract.py:68]: Can you also include the original ARC-69 note JSON metadata here?(https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0069.md)

Comment (D0rs4n) [reply smart_contracts/fractdistribution/contract.py]: done

**Review by visdauas — APPROVED** [review]
LGTM

**Pull request merged**